# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/build/Omnijar.cpp
# Commit: 24ee1c0756ae
# Full Hash: 24ee1c0756ae49b467d53c499dd0c16839e05eb2
# Author: Randell Jesup <rjesup@wgate.com>
# Date: 2022-03-08 22:01:59
# Regressor Bug: 1744043
# File Overlap Count: 1
# Description:
#   Bug 1744043: Clean up nsJAR r=nika,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132794
# ==============================================================================

diff -r 1c746ea21b95 -r 24ee1c0756ae xpcom/build/Omnijar.cpp
--- a/xpcom/build/Omnijar.cpp	Tue Mar 08 13:05:28 2022 +0000
+++ b/xpcom/build/Omnijar.cpp	Tue Mar 08 13:57:56 2022 +0000
@@ -26,11 +26,9 @@
 
 void Omnijar::CleanUpOne(Type aType) {
   if (sReader[aType]) {
-    sReader[aType]->CloseArchive();
     sReader[aType] = nullptr;
   }
   if (sOuterReader[aType]) {
-    sOuterReader[aType]->CloseArchive();
     sOuterReader[aType] = nullptr;
   }
   sPath[aType] = nullptr;
@@ -77,8 +75,8 @@
     return;
   }
 
-  RefPtr<nsZipArchive> zipReader = new nsZipArchive();
-  if (NS_FAILED(zipReader->OpenArchive(file))) {
+  RefPtr<nsZipArchive> zipReader = nsZipArchive::OpenArchive(file);
+  if (!zipReader) {
     return;
   }
 
@@ -87,8 +85,8 @@
   if (NS_SUCCEEDED(nsZipHandle::Init(zipReader, MOZ_STRINGIFY(OMNIJAR_NAME),
                                      getter_AddRefs(handle)))) {
     outerReader = zipReader;
-    zipReader = new nsZipArchive();
-    if (NS_FAILED(zipReader->OpenArchive(handle))) {
+    zipReader = nsZipArchive::OpenArchive(handle);
+    if (!zipReader) {
       return;
     }
   }
